shakedown/dcos/task.py

```diff
@@ -11,28 +11,40 @@
 
 def get_tasks(task_id='', completed=True):
     """ Get a list of tasks, optionally filtered by task id.
+        The task_id can be the abbrevated.  Example: If a task named 'sleep' is
+        scaled to 3 in marathon, there will be be 3 tasks starting with 'sleep.'
 
         :param task_id: task ID
         :type task_id: str
         :param completed: include completed tasks?
         :type completed: bool
 
-        :return: a tuple of tasks
-        :rtype: tuple
+        :return: a list of tasks
+        :rtype: []
     """
 
     client = mesos.DCOSClient()
     master = mesos.Master(client.get_master_state())
-    tasks = master.tasks(completed=completed, fltr=task_id)
+    mesos_tasks = master.tasks(completed=completed, fltr=task_id)
+    tasks = [task.__dict__['_task'] for task in mesos_tasks]
 
-    return tasks
+    return [task for task in tasks if task['id'].startswith(task_id)]
```

Comment:
while questionable :) I switched to `startswith` so that if you have a `sleep` app in marathon and scale to 3, it will create 3 tasks with 3 task ids starting with `sleep.` followed by unique chars.   this provides the ability to `get_tasks('sleep.')` and get all of them. 
The prior implimentation was all or nothing... either "" or the exact task_id.  This is an improvement.  I wasn't thinking regex and didn't want to add any complexity to it.  if you think it is worth it... I can do it.